LambdaEngine/Include/Containers/TArray.h

```diff
@@ -1,10 +1,1210 @@
 #pragma once
-#include <vector>
+#include "TUtilities.h"
+
+#include <iterator>
 
 // Disable the DLL- linkage warning for now
 #ifdef LAMBDA_VISUAL_STUDIO
 	#pragma warning(disable : 4251)
 #endif
 
-template <typename T>
-using TArray = std::vector<T>;
\ No newline at end of file
+namespace LambdaEngine
+{
+	/*
+	* Dynamic Array similar to std::vector
+	*/
+	template<typename T>
+	class TArray
```

Comment:
Since you've mostly been naming containers following the pattern: `std::container` -> `TContainer`, one might assume that `TArray` is `std::array`. I suggest renaming `TArray` to `TVector`.

Reply:
I think that this quote from the creator of C++ explains pretty good why I used this naming convention. "_One could argue that valarray should have been called vector because it is a traditional mathematical vector and that vector should have been called array._ "

I.e., vector is a terrible name for this data structure, and that's why the name is TArray since it better describes what it is.  When even the creator suggests that the name is terrible, why should we make the same mistake?

If we would like to have a data structure similar to std::array in the future, I suggest the name TStaticArray or just a template specialization of TArray that uses a compile-time specified size instead. 

And if there should be a name change, it would make more sense to name it TDynamicArray instead since it better describes what it is, compared to TVector. Vector is confusing when you also have mathematical datatypes (glm::vec3, etc.) in your application, and when this datatype has nothing to do with real vectors.

Since the only reason for creating your own data structures is to beat the standard implementation (Or as a learning experience, of course), I find it weird to implement their mistakes (Yes I find the name std::vector a mistake). 

I can agree to change the name, but I'm afraid I have to disagree with calling it TVector because the standard library does so since we are not building a standard library. Sure, people might be more familiar with the vector term, but many implementations do not call their arrays for vector, and I think that we should be one of those. 

But this is just my option.